### PR TITLE
UnixPB: prevent improper hostname and /etc/hosts change

### PR DESCRIPTION
**Pull request by Haroon-Khel**
Ive added a condition to prevent the playbooks from improperly changing the hostname or adding an incorrect line into /etc/hosts when the ip address is passed as the `inventory_hostname` variable

### CODE REVIEW

**Review by Willsparker — COMMENTED** [review]
While I'm not opposed to this particular change to stop the _incorrect_ entry going into `/etc/hosts`, this PR doesn't solve the issue of there still being a missing host entry (I'm assuming this PR is related to the[ Slack thread?](https://adoptopenjdk.slack.com/archives/C53GHCXL4/p1602600149107700)). 
An additional task could be put in to say 
```
if [[ inventory_hostname == ansible_default_ipv4.address ]] then; 
  lookup_correct_hostname
```
however, I'll admit off the top of my head, I don't have any suggestions on how to do that.

**Comment by Haroon-Khel** [comment]
@Willsparker I agree. But I would still like to prevent improper entries within the /etc/hosts file, hence this pr.
Im looking into a way to pass the machine name to ansible via a command line parameter

**Review by sxa — CHANGES_REQUESTED** [review]
Changing my review:
1. Can we print out a message saying "Not updating hostname" in this situation
2. Can we also prevent it being updated when the hostname says `localhost`?

The other thing we could do is explicitly ensure there is an extra for whatever the output of `hostname` is on the machine in the situation where we are not updating it, which would resolve the underlying problem that lead to this PR

**Comment by Haroon-Khel** [comment]
It seems that you can pass the machine name to the playbooks using
```
--extra-vars "inventory_hostname = foo"
```
This overrides the ansible variable `inventory_hostname`

**Comment by Haroon-Khel** [comment]
If the machine name isnt passed as a variable via the `--extra-vars` method, the `inventory_hostname` variable defaults to the ip address of the machine (or whatever your preferred method of making ansible target the remote machine is), and then this passes through the new conditions

**Review by sxa — CHANGES_REQUESTED** [review]
Minor adjustment to the message to make it easier for people to figure out how to do the right thing in the future. I haven't verified that the suggestion works (you never know with `<>` signs!

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml [hunk 1]
```diff
@@ -66,9 +75,21 @@
...
+    - inventory_hostname != ansible_default_ipv4.address
+    - inventory_hostname != "localhost"
   tags: hosts_file
 
+- debug:
+    msg: "Inventory_hostname is the same as the ip address or is localhost.
+    The fqdn has not been added to /etc/hosts.
+    The hostname of the machine has not been updated.
+    Manually add the fqdn to /etc/hosts.
+    Manually update the hostname in /etc/hostname."
```
[the remark this answers is not in the document]
Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/main.yml]: Ive wrapped `inventory_hostname=<realhostname>` in single quote marks. Ansible doesnt seem to have a problem with it, and this is how it should be used on the command line (double quotes can also be used, but ansible doesnt like the use of double quotes within the debug message)

**Review by Willsparker — APPROVED** [review]

**Review by sxa — APPROVED** [review]

**Pull request merged**